features/appmgt/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/apis/v1/lifecycle_api_router.jag

```diff
@@ -131,364 +131,405 @@ require('/modules/publisher.js').exec(function (ctx) {
+            return;
+        }
+
+        var userManager = server.userManager(server.current(session).tenantId);
+        var user = server.current(session);
+        var permission = null;
+        if (appType == 'webapp') {
+            permission = config.permissions.webapp_publish;
+        } else if (appType == 'mobileapp') {
+            permission = config.permissions.mobileapp_publish;
+        }
+        // check whether current user has publisher permissions
+        var hasPublisherPermission = permissions.isAuthorized(user.username, permission, userManager);
+        if (!hasPublisherPermission) {
+            log.error("Insufficient permissions to perform '" + action + "' on " + appType);
+            response.status = 401;
+            response.contentType = "application/json";
+            content.status = "Access Denied";
+            content.messages.push("You do not have publisher permissions to perform '"
+                                  + action + "' on " + appType + "s");
+            response.content = stringify(content);
+            return;
+        }
+
+        var carbon = require('carbon');
+        var artifactManager = rxtManager.getSystemArtifactManager(appType, session);
+
+        try {
+            var currentArtifact = artifactManager.get(uuid);
+        } catch (e) {
+            // this exception is thrown by
+            // GenericArtifactManager.getGenericArtifact(id) throws GovernanceException
+            log.error("The asset with UUID " + uuid + " could not be retrieved.");
+            log.error(e);
+            response.status = 404;
+            response.contentType = "application/json";
+            content.status = "Not Found";
+            content.messages.push("The " + appType + " with UUID " + uuid + " cannot be retrieved");
+            response.content = stringify(content);
+            return;
+        }
+
+        // grant permissions
+        var roles = (context.content) ? context.content.roles : null;
+        if (roles && (roles.length > 0)) {
+            var currentRole = null;
+            try {
+                roles.forEach(function (role) {
+                    currentRole = role;
+                    userManager.authorizeRole(currentRole, currentArtifact.path,
+                                              carbon.registry.actions.GET);
+                });
+            } catch (e) {
+                log.error("Cannot grant permissions to role '" + currentRole
+                          + "' to perform action '" + carbon.registry.actions.GET + "'");
+                log.error(e);
+                // continue work flow
+            }
+        }
+
+        // check permissions to perform life cycle action
+        try {
+            if (!permissions.isLCActionsPermitted(user.username, currentArtifact.path,
+                                                  userManager)) {
+                log.error("Insufficient permissions to perform '" + action + "' on " + appType
```

Comment:
messages formatted, see commit 42edb3b60ee62d4da43658986e3fb68d2b0a61e4 of PR #185